backend/psql/room.go

```diff
@@ -38,18 +38,41 @@ type RoomBinding struct {
 
 func (rb *RoomBinding) GetMessage(ctx scope.Context, id snowflake.Snowflake) (*proto.Message, error) {
 	var msg Message
-	err := rb.DbMap.SelectOne(
+
+	nDays, err := rb.DbMap.SelectInt("SELECT retention_days FROM room WHERE room = $1", rb.Name)
+	if err != nil {
+		return nil, err
+	}
+
+	var query string
+	args := []interface{}{rb.Name, id.String()}
+
+	if nDays == 0 {
```

Comment:
Rather than duplicating the query, let's just fetch the message and check its timestamp on the server side.